src/olympia/amo/management/commands/check_locales_completion_rate.py

```diff
@@ -0,0 +1,175 @@
+import re
+
+import hashlib
+import requests
+from collections import defaultdict
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from django.template import loader
+
+from olympia.amo.utils import send_mail
+from olympia.core.languages import ALL_LANGUAGES
+
+
+class Command(BaseCommand):
+    help = (
+        'Check locales completion rate for AMO projects in Pontoon and warn '
+        'us by email of any falling below pre-established thresholds'
+    )
+    PONTOON_API = 'https://pontoon.mozilla.org/graphql'
+    PONTOON_PROJECTS = ('amo', 'amo-frontend')
+    QUERY_TEMPLATE = """
+        {
+          project(slug: "%(project)s") {
+            name,
+            localizations {
+              locale {
+                code,
+                name
+              }
+              totalStrings,
+              approvedStrings
+            }
+          }
+        }
+    """
+    # 40% completion is the first step of the project. Then we'll move it up to
+    # 80% for a limited period, and eventually it will be set to 70% to account
+    # for new strings being added over time.
+    COMPLETION_THRESHOLD = 40
+    # Arbitrary list of locales we always want to keep, either because they
+    # receive the most traffic/affect the most users, or because they are RTL
+    # and we want to keep at least one of those to ensure AMO works in RTL.
+    LOCALES_TO_ALWAYS_KEEP = (
+        'ar',
+        'de',
+        'en-US',
+        'es-ES',
+        'fr',
+        'he',
+        'it',
+        'ja',
+        'pl',
+    )
+    EMAIL_RECIPIENTS = ('amo-locales-notifications@mozilla.com',)
+    EMAIL_SUBJECT = 'AMO locales completion rate check'
+    EMAIL_TEMPLATE_PATH = 'amo/emails/locales_completion.ltxt'
+
+    def add_arguments(self, parser):
+        """Handle command arguments."""
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            default=False,
+            help='Fetch data from Pontoon but do not send emails.',
+        )
+
+    def handle(self, *args, **options):
+        locales_below_threshold, locales_above_threshold = (
+            self.find_locales_below_and_above_threshold()
+        )
+        self.warn_about(
+            locales_below_threshold=locales_below_threshold,
+            locales_above_threshold=locales_above_threshold,
+        )
+
+    def find_locales_below_and_above_threshold(self):
+        locales_below_threshold = set()
+        locales_above_threshold = set()
+        seen_locales = defaultdict(int)
+        for project in self.PONTOON_PROJECTS:
```

Comment:
GraphQL also typically supports multiple queries in a single request. There are only 2 projects but that could be less prone to random errors and _technically_ more efficient.

```python
QUERY = """
query {
  amo: project(slug: "amo") {
    ...projectFields
  }
  amoFrontend: project(slug: "amo-frontend") {
    ...projectFields
  }
}

fragment projectFields on Project {
  name
  localizations {
    locale {
      code
      name
    }
    totalStrings
    approvedStrings
  }
}
"""
```

That would probably do the trick.